Fantasy Battle Game/Assets/Scripts/Player/Champions/ChampionsUtilities/Movements/HexMovement.cs

```diff
@@ -139,22 +145,106 @@ public IEnumerator LookAtIEnumerator(Vector3 point)
             orientation_ = transform.localRotation.eulerAngles.y;
         }
 
-        protected void CalculateRoute()
+        private double EstimateCost(Tile current, Tile destination)
         {
+            var dx = Math.Abs(current.Coordinate.FirstCoord - destination.Coordinate.FirstCoord);
+            var dy = Math.Abs(current.Coordinate.SecondCoord - destination.Coordinate.SecondCoord);
+
+            return 1 * (dx + dy) + (Math.Sqrt(2) - 2 * 1) * Math.Min(dx, dy);
+        }
+
+        protected void CalculateRoute(int kindOfAlgorithm)
+        {
+            
             route_.Clear();
-            Tile TempDestinationTile = currentTile;
-            route_.Add(TempDestinationTile);
-            while (TempDestinationTile != DestinationTile)
+                                
+            openList.Clear();
+            closedList.Clear();
+
+            switch (kindOfAlgorithm)
             {
-                var list = SortList(TempDestinationTile, DestinationTile);
+                case 1:
```

Comment:
unclear meaning of variable, magic numbers are not needed in code. Have you considered using enum?